### PR TITLE
Updating Code Definitions w/ Ids

### PR DESCRIPTION
**Pull request by marlenc3**
majority of this update is updating tests -- but should be passing up all ids now

### CODE REVIEW

File: src/vellum/workflows/events/types.py [hunk 1]
```diff
@@ -28,6 +28,15 @@ def serialize_type_encoder(obj: type) -> Dict[str, Any]:
     }
```
Comment (marlenc3) [inline src/vellum/workflows/events/types.py:28]: Main change is this file 🔥

File: src/vellum/workflows/events/node.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class _BaseNodeExecutionBody(UniversalBaseModel):
```
Comment (marlenc3) [inline src/vellum/workflows/events/node.py:21]: event bodies will also carry their ids -- let me know if this is not something we want

File: src/vellum/workflows/events/workflow.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ class _BaseWorkflowExecutionBody(UniversalBaseModel):
 
     @field_serializer("workflow_definition")
     def serialize_workflow_definition(self, workflow_definition: Type, _info: Any) -> Dict[str, Any]:
-        return serialize_type_encoder(workflow_definition)
+        return serialize_type_encoder_with_id(workflow_definition)
```
Comment (marlenc3) [inline src/vellum/workflows/events/workflow.py:32]: workflow bodies to carry their ids

File: src/vellum/workflows/events/node.py (hunk 2 above)
Comment (dvargasfuertes) [reply src/vellum/workflows/events/node.py:21]: this is something we need to replace the translator logic, so def keep

File: src/vellum/workflows/events/tests/test_event.py [hunk 4]
```diff
@@ -73,6 +74,7 @@ class MockWorkflow(BaseWorkflow[MockInputs, BaseState]):
                 "name": "workflow.execution.initiated",
                 "body": {
                     "workflow_definition": {
+                        "id": str(uuid4_from_hash(MockWorkflow.__qualname__)),
```
Comment (dvargasfuertes) [inline src/vellum/workflows/events/tests/test_event.py]: [nit] slight pref for setting these two as hardcoded files in the top of the file and using in tests

File: src/vellum/workflows/events/types.py [hunk 5]
```diff
@@ -46,9 +55,17 @@ def encode(obj: type) -> "CodeResourceDefinition":
         return CodeResourceDefinition(**serialize_type_encoder(obj))
 
 
+class VellumCodeResource(CodeResourceDefinition):
+    id: UUID
```
Comment (dvargasfuertes) [inline src/vellum/workflows/events/types.py]: [nit] ah I think _here_, we're safe to add directly to `CodeResourceDefinition`. It's on the vellum side that we'll want to split out into a separate class

**Review by dvargas92495 — APPROVED** [review]

File: src/vellum/workflows/events/types.py (hunk 5 above)
Comment (marlenc3) [reply src/vellum/workflows/events/types.py]: oh bet! that'll be a fast change

**Pull request merged**